src/main/java/dev/j3fftw/litexpansion/Events.java

```diff
@@ -161,101 +158,98 @@ public void onDye(PlayerInteractEntityEvent e) {
-                Bukkit.getServer().getPluginManager().callEvent(newEvent);
-                block.setType(Material.AIR);
-                e.getPlayer().playSound(blockLocation, Sound.BLOCK_STONE_BREAK, 1.5F, 1F);
-
-                // This has to be done because the item in the main hand is not a pickaxe
-                if (blockType == Material.STONE) {
-                    blockLocation.getWorld().dropItem(blockLocation,
-                        new ItemStack(Material.COBBLESTONE)
-                    );
-
-                } else {
-                    blockLocation.getWorld().dropItem(blockLocation,
-                        new ItemStack(blockType)
-                    );
-                }
+            Validate.notNull(diamondDrill);
+            if (!Check(diamondDrill, event, blockLocation)) {
+                return;
+            }
+
+            if (!Tag.MINEABLE_PICKAXE.isTagged(blockType)) {
+                return;
             }
+
+            DrillUse(1.5f, block, blockType, blockLocation, event);
         }
     }
 
-    //TODO combine onDiamondDrillUse with onMiningDrillUse
-    @EventHandler
-    public void onDiamondDrillUse(PlayerInteractEvent e) {
-        final Block block = e.getClickedBlock();
-        if (block == null) {
+
+    public boolean Check(MiningDrill miningDrill, PlayerInteractEvent event, Location blockLocation) {
+        return miningDrill.isItem(event.getItem())
+            && !miningDrill.isDisabled()
+            && Slimefun.getProtectionManager().hasPermission(event.getPlayer(),
+            blockLocation, Interaction.BREAK_BLOCK);
+    }
+
+    public void DrillUse(float charge, Block block, Material blockType,
+                         Location blockLocation, PlayerInteractEvent event
+    ) {
+        event.setCancelled(true);
+
+        final SlimefunItem slimefunItem = BlockStorage.check(block);
+
+        if (slimefunItem != null) {
             return;
         }
 
-        final Material blockType = block.getType();
-        final Location blockLocation = block.getLocation();
+        final Rechargeable item = (Rechargeable) SlimefunItem.getByItem(event.getItem());
 
-        final MiningDrill diamondDrill = (MiningDrill) SlimefunItem.getById(Items.DIAMOND_DRILL.getItemId());
+        if (item == null) {
+            return;
+        }
 
+        if (!item.removeItemCharge(event.getItem(), charge)) {
+            return;
+        }
 
-        Validate.notNull(diamondDrill, "Can not be null");
-        if (diamondDrill.isItem(e.getItem())
-            && !diamondDrill.isDisabled()
-            && SlimefunTag.MINEABLE_PICKAXE.isTagged(blockType)
-            && Slimefun.getProtectionManager().hasPermission(e.getPlayer(),
-            blockLocation, Interaction.BREAK_BLOCK)
-        ) {
-            e.setCancelled(true);
+        BlockBreakEvent newEvent = new BlockBreakEvent(block, event.getPlayer());
+        Bukkit.getServer().getPluginManager().callEvent(newEvent);
```

Comment:
I know you weren't before, but shouldn't this be checking to see if the event was cancelled?